target/linux/ramips/dts/mt7621_snr_snr-cpe-me2-lite.dts

```diff
@@ -0,0 +1,164 @@
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "snr,snr-cpe-me2-lite", "mediatek,mt7621-soc";
+	model = "SNR-CPE-ME2-Lite";
+
+	leds {
+		compatible = "gpio-leds";
+
+		sys {
+			label = "green:sys";
+			gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		vpn {
+			label = "green:vpn";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "jtag", "wdt";
+		function = "gpio";
+	};
+};
+
+&gdma {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <104000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "config";
+				reg = <0x30000 0x10000>;
+				read-only;
```

Comment:
If it is uboot-env then you can remove the r/o flag.
It is also recommended to add the uboot-env save area to `package/boot/uboot-envtools/files/ramips`, and the `uboot-envtools` package.

Reply:
This is a section with nvram variables of the factory firmware, not uboot-env.

Reply:
Ok, but then at what address (and area size) will u-boot save with the `saveenv` command? This is what you need to specify in `package/boot/uboot-envtools/files/ramips`

Reply:
There is no separate section for uboot-env. The environment occupies one whole flash sector, which is "embedded" in the text segment with the U-Boot code.

Reply:
> There is no separate section for uboot-env. The environment occupies one whole flash sector, which is "embedded" in the text segment with the U-Boot code.

This is clear from the previous comment.

> This is a section with nvram variables of the factory firmware, not uboot-env.

To understand the save area of uboot-env, backup mtd0 "u-boot", change the `setenv` parameter to uboot, save it to `saveenv` (here you can see the size of the area), backup mtd0 again and you will see the save address of uboot-env - this will be search area (starting address and size) which must be specified in the `package/boot/uboot-envtools/files/ramips` file.